### PR TITLE
AWS Batch job control

### PR DESCRIPTION
**Pull request by tsibley**
See commit messages for details, and also the [new changelog entries](https://github.com/nextstrain/cli/pull/308/commits/e3888cd527b694ca619443943c01f66696982a01).

This work is motivated by deeper integration for GitHub Actions workflows that launch builds as AWS Batch jobs.

## Checklist

- [x] Checks pass

### CODE REVIEW

**Comment by tsibley** [comment]
Repushed with some minor refinements re: the clarity of code around `job.stopped` vs. `job.stop_sent` and the UX around `--cancel`.

File: nextstrain/cli/command/build.py [hunk 1]
```diff
@@ -43,6 +43,13 @@ def register_parser(subparser):
                  "Currently only supported when also using :option:`--aws-batch`.",
         action = "store_true")
 
+    parser.add_argument(
+        "--detach-on-interrupt",
+        help   = "Detach from the build when an interrupt (e.g. :kbd:`Control-C` or ``SIGINT``) is received.  "
+                 "Interrupts normally cancel the build (when sent twice).  "
```
Comment (joverlee521) [inline nextstrain/cli/command/build.py:49]:
```suggestion
                 "Interrupts normally cancel the build (when sent twice if stdin is a terminal).  "
```

**Review by joverlee521 — APPROVED** [review]
I really like the new `--cancel` flag, that will be really useful for stopping trial builds 🌟 

I'm a little unsure about the expected use of the `--detach-on-interrupt` option in the context of GitHub Action managed builds. This is definitely due to my lack of understanding of signals and how signals would propagate through GitHub Actions. (Still processing this [GitHub community forum](https://github.com/orgs/community/discussions/26311) and this [demo of GitHub actions signal handling](https://github.com/ringerc/github-actions-signal-handling-demo).)

Can definitely test this out myself later, but wanted to leave these questions:
* Would `--detach-on-interrupt` be used to chain the jobs to work around the 6hr job limit? 
* Does canceling a GitHub Action workflow also cancel the attached AWS Batch job?

**Comment by tsibley** [comment]
Repushed to add suggested clarification to option help and resolve conflicts in the changelog by rebasing onto latest `master`.

**Comment by tsibley** [comment]
> I'm a little unsure about the expected use of the `--detach-on-interrupt` option in the context of GitHub Action managed builds. This is definitely due to my lack of understanding of signals and how signals would propagate through GitHub Actions. (Still processing this [GitHub community forum](https://github.com/orgs/community/discussions/26311) and this [demo of GitHub actions signal handling](https://github.com/ringerc/github-actions-signal-handling-demo).)

Yeah, it's kind of a mess. Signal use/propagation in GitHub Actions is lightly documented but not comprehensively.

We can do our own propagation if necessary, e.g. to signal the whole process group instead of just the leader. I [did this in previous WIP prototypes](https://github.com/nextstrain/.github/commit/0b0cf269797a03c963037af9670a117bd570f959#diff-33ef03d95073a428895165acc844958628ca784b46ea992eb748aa2f754239f1R287-R304).

> > * Does canceling a GitHub Action workflow also cancel the attached AWS Batch job?
> 
> Yes, without `--detach-on-interrupt` given and with this PR's change to only require a single Ctrl-C/SIGINT when stdin is not a tty. Not currently.

Realizing this "yes" might still be modulated by signal propagation.

**Comment by tsibley** [comment]
> Yeah, it's kind of a mess.

Though to be fair to GitHub Actions, the signalling cancellation in any system can be messy, esp. when there's support for graceful cancellation, and hard to simplify.

**Pull request merged**